src/axolotl/cli/utils/lora_merge.py

```
@@ -856,6 +877,229 @@ def _detect_per_expert_unfused_mismatch(model_shards, lora_state) -> bool:
+                with safetensors.safe_open(shard, framework="pt") as f:
+                    for k in f.keys():
+                        if _PER_EXPERT_WEIGHT_RE.search(k) and ".experts." in k:
+                            if f.get_slice(k).get_dtype() == "U8":
+                                return True
+            else:
+                tensors = torch.load(shard, map_location="meta", weights_only=True)  # nosec B614
+                for k, t in tensors.items():
+                    if _PER_EXPERT_WEIGHT_RE.search(k) and t.dtype == torch.uint8:
+                        return True
+        except Exception:  # noqa: BLE001  # nosec B112 - unreadable shard: skip the peek, not fatal
+            continue
```

Ruff (0.15.20)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)